helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/WagedRebalancer.java

```diff
@@ -326,32 +328,17 @@ public void close() {
     return newIdealStates;
   }
 
-  // Coordinate baseline recalculation and partial rebalance according to the cluster changes.
+  // Coordinate global rebalance and partial rebalance according to the cluster changes.
   protected Map<String, ResourceAssignment> computeBestPossibleAssignment(
```

Comment:
The method body only consists of 2 lines, do you make it protected just for the purpose of unit testing? 
Otherwise, the method can be just inside the caller method.
For unit testing,  I think there's other alternative approaches

Reply:
This is mainly for the dry-run method that we are using in the verifier.
Any better suggestion without introducing duplicate code is welcome. Could you please be more elaborated?

Reply:
I think you could make the globalRebalance method protected and override the globalRebalance in your dry-run class (not updating the metadata store). The method of only 2 lines can be avoided

Reply:
It's debatable. But I still prefer the current way.
1. A method with 2 lines is not a concern to me as long as the logic itself makes sense. For here, this method include the WAGED rebalancer's core logic. It is a standalone and self containing logic.
Imaging if I unwrap the globalRebalance and partialRebalance method here, we will have more lines in this method. But it does not do anything good.
2. As you suggested, then I will need to override 2 methods instead of one. And the testing logic for verifying the WAGED core algorithm has to call these 2 methods separately in each test cases.

Reply:
The second suggestion is using a override subclass of metaDataAssignment as the passed instance in the dry-run class. E.g,
```
MetaDataAssignmentDataStore fakeMetaDataAssignmentDataStore = new XXX {
   @override
    boolean writeToDB() {
      // do nothing
    }
}
class DryRunRebalancer {
   DryRunRebalancer() {
    super(fakeMetaDataAssignment)
   }
}
```
Will it work? I don't think you will need to override any methods if your dry-run is trying to get the result and not persisting it at all

Reply:
Sorry I should have made it clearer. I already did that in the dryrun rebalancer. But overriding the assignment datastore is not enough. I don't need to calculate anything in a dryrun rebalancer. That is for the verifier to use. So I just need to read from the persisted assignment.
In this case, overriding the computing methods is required. Otherwise, we run the risk that the verifier will calculate different results from the remote controller. Note the waged rebalancer algorithm is not static as the CrushEd does.